mlibrary/arclight
# DCP2-378 - trap modal focus

Added `app/assets/javascripts/blacklight/modal.js`:

* because arclight can bring up multiple modals (LATER), track modals in `ACTIVE_MODALS`
* use `shown.bs.modal` to hide the first modal when the second modal has been activated
* use `loaded.blacklight.blacklight-modal` to maintain focus when the contents of the modal are updated via AJAX
* keep focus on next/prev links when user is paging through modal contents (via the `lastActive` tracking)

To test:

* open the advanced search modal: keyboard navigation should never leave the modal
* open a "more" modal from a filter:
  * keyboard navigation should never leave the modal
  * clicking next/prev should keep the focus on next/prev links
  * clicking one of the short options will set the focus to the close button
